libc/CMakeLists.txt

```diff
@@ -8,6 +8,13 @@ endif()
 include(${LLVM_COMMON_CMAKE_UTILS}/Modules/CMakePolicy.cmake
   NO_POLICY_SCOPE)
 
+# If we are not building as a part of LLVM, build libc as an
+# standalone project, using LLVM as an external library:
+if(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
+  project(libc)
+  set(LIBC_BUILT_STANDALONE TRUE)
+endif()
```

Comment:
Given that `libc` doesn't support builds rooted in the `libc` directory (and I don't think we want to add support for it), maybe we should be erroring out here instead.

Reply:
I don't think so, would we stop supporting standalone builds of everything else (clang, llvm, bolt, lldb, etc.) unless they use `runtimes`? I don't really think an error is necessary.

Reply:
I think a warning (or error) would be a very user-friendly feature to add. Then we can point users in the right direction if they make a mistake.

Reply:
Opened #119426 for this.